sdk/python/feast/errors.py

```diff
@@ -78,6 +78,16 @@ def __init__(self, offline_store_name: str, data_source_name: str):
         )
 
 
+class FeatureNameCollisionError(Exception):
+    def __init__(self, feature_refs_collisions: List[str]):
+        feature_name_collisions = [ref.split(":")[1] for ref in feature_refs_collisions]
+        feature_names = ", ".join(x for x in feature_name_collisions)
+        super().__init__(
+            f"The following feature name(s) have collisions: {feature_names}. Set 'full_feature_names' "
```

Comment:
We should definitely talk about the feature view/table names that have collided.